simd/src/Kokkos_SIMD_Common.hpp

```diff
@@ -308,44 +332,52 @@ KOKKOS_FORCEINLINE_FUNCTION where_expression<M, T>& operator/=(
 // fallback implementations of reductions across simd_mask:
 
 template <class T, class Abi>
-[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION bool all_of(
-    simd_mask<T, Abi> const& a) {
+[[nodiscard]] KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION std::enable_if_t<
+    !std::is_same_v<Abi, Kokkos::Experimental::simd_abi::scalar>, bool>
+all_of(simd_mask<T, Abi> const& a) {
   return a == simd_mask<T, Abi>(true);
 }
 
 template <class T, class Abi>
-[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION bool any_of(
-    simd_mask<T, Abi> const& a) {
-  return a != simd_mask<T, Abi>(false);
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION std::enable_if_t<
+    std::is_same_v<Abi, Kokkos::Experimental::simd_abi::scalar>, bool>
```

Comment:
We are not consistent.  Here using SFINAE and in `where` above providing a more specialized overload.  (I prefer the more specialized overloads moved into the `<Kokkos_SIMD_Scalar.hpp>` header)